analyzers/tests/SonarAnalyzer.UnitTest/Wrappers/ObjectCreationFactoryTest.cs

```diff
@@ -0,0 +1,100 @@
+            wrapper.Initializer.Should().BeEquivalentTo(objectCreation.Initializer);
+            wrapper.ArgumentList.Should().BeEquivalentTo(objectCreation.ArgumentList);
+            wrapper.TypeAsString(compilation.GetSemanticModel(syntaxTree)).Should().Be("A");
+        }
+
+        [TestMethod]
+        public void ImplicitObjectCreationSyntax()
+        {
+            var compilation = CreateCompilation("public class A{public int X;public A(int y){}} public class B{void Foo(){A bar =new(1){X = 2};}}");
+            var syntaxTree = compilation.SyntaxTrees.First();
+            var objectCreation = (ImplicitObjectCreationExpressionSyntaxWrapper)syntaxTree.GetRoot().DescendantNodes().First((node => node.IsKind(SyntaxKindEx.ImplicitObjectCreationExpression)));
+            var wrapper = ObjectCreationFactory.Create(objectCreation);
+            wrapper.Expression.Should().BeEquivalentTo(objectCreation.SyntaxNode);
+            wrapper.Initializer.Should().BeEquivalentTo(objectCreation.Initializer);
+            wrapper.ArgumentList.Should().BeEquivalentTo(objectCreation.ArgumentList);
+            wrapper.TypeAsString(compilation.GetSemanticModel(syntaxTree)).Should().Be("A");
+        }
+
+        [TestMethod]
+        public void GivenImplicitObjectCreationSyntaxWithMissingType_HasEmptyType()
+        {
+            var compilation = CreateCompilation("public class B{void Foo(){var bar =new();}}");
+            var syntaxTree = compilation.SyntaxTrees.First();
+            var objectCreation = (ImplicitObjectCreationExpressionSyntaxWrapper)syntaxTree.GetRoot().DescendantNodes().First((node => node.IsKind(SyntaxKindEx.ImplicitObjectCreationExpression)));
+            var wrapper = ObjectCreationFactory.Create(objectCreation);
+            wrapper.TypeAsString(compilation.GetSemanticModel(syntaxTree)).Should().BeEmpty();
+        }
+
+        [TestMethod]
+        public void GivenNull_ThrowsException()
+        {
+            Action action = () => { ObjectCreationFactory.Create(null); };
+#if NET
+            action.Should().Throw<ArgumentNullException>().WithMessage("Argument should not be null (Parameter 'node')");
+#else
+            action.Should().Throw<ArgumentNullException>().WithMessage("Argument should not be null\nParameter name: node");
+#endif
+        }
+
+        [TestMethod]
+        public void GivenNonConstructor_ThrowsException()
+        {
+            var compilation = CreateCompilation("public class A{}");
+            var syntaxTree = compilation.SyntaxTrees.First();
+            var classDeclaration = syntaxTree.GetRoot().DescendantNodes().OfType<ClassDeclarationSyntax>().First();
+            Action action = () => { ObjectCreationFactory.Create(classDeclaration); };
+            action.Should().Throw<InvalidOperationException>().WithMessage("Unexpected type: ClassDeclarationSyntax");
+        }
+
+        private static CSharpCompilation CreateCompilation(string code) =>
```

Comment:
This shouldn't be needed. You an use either `SnippetCompiler` or `TestHelper.Compile` to achieve the same with existing helpers.